Gravity-tube/src/Reducers/PlaylistReducer.js

```diff
@@ -0,0 +1,35 @@
+import React from 'react'
+
+function playlistReducer(state,{type,payload}) {
+  switch (type) {
+      case "SET_NEW_PLAYLIST":
+          return {...state, playlists: payload}
+      case "ADD_VIDEO_DATA_IN_PLAYLIST":
+          return  {
+            ...state,
+            playlists: state.playlists.map((item) =>
+              item._id === payload._id
+                ? { ...item, videos: payload.videos }
+                : item
+            ),
+          };
+      case "REMOVE_PLAYLIST":
+          return  {
+            ...state,
+            playlists: payload
+          };
```

Comment:
- you can use your first case `SET_NEW_PLAYLIST` case here also instead of make new case